File: EASendMail/Samples_ASPNetMvc/C#/WebProject1/WebProject1/Oauth/OauthProviderInterface.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace EASendMail.Oauth
{
    public interface OauthProviderInterface
    {
        string ClientId { get; set; }
        string ClientSecret { get; set; }
        string UserEmail { get; set; }
        string AccessToken { get; set; }
        string RefreshToken { get; set; }
        string RedirectUri { get; set; }
        string Scope { get; set; }
        string AuthUri { get; set; }
        string TokenUri { get; set; }

        int TokenExpiresInSeconds { get; set; }
        DateTime AccessTokenTimeStamp { get; set; }

        string GenerateFullAuthUri();
        string GenerateAccessTokenRequestData(string authorizationCode);
        string GenerateRefreshTokenRequestData();

        void ClearToken();
    }
}

File: EASendMail/Samples_ASPNetMvc/VB/WebProject1/WebProject1/Oauth/OauthProviderInterface.vb
Public Interface OauthProviderInterface
    Property ClientId As String
    Property ClientSecret As String
    Property UserEmail As String
    Property AccessToken As String
    Property RefreshToken As String
    Property RedirectUri As String
    Property Scope As String
    Property AuthUri As String
    Property TokenUri As String
    Property TokenExpiresInSeconds As Integer
    Property AccessTokenTimeStamp As DateTime
    Function GenerateFullAuthUri() As String
    Function GenerateAccessTokenRequestData(ByVal authorizationCode As String) As String
    Function GenerateRefreshTokenRequestData() As String
    Sub ClearToken()
End Interface
